DOCS-1472: Use case page train and deploy a classification model (#2258)
viamrobotics/docs: docs/use-cases/deploy-ml.md

linkTitle: "Train and deploy classification models"
weight: 50
type: "docs"
tags: ["data management", "data", "services"]
no_list: true
description: "Use Viam's machine learning capabilities to train image classification models and deploy these models to your machines."
image: "/ml/training.png"
imageAlt: "Machine Learning"
images: ["/ml/training.png"]
---

> Deploy an image classification model onto your machine with Viam's machine learning (ML) service.

Author: @andf-viam Added this intro-- LMK what you think! struggling to add anything for next steps though-- liked your suggestions but data management and view image streams, they've already done, I can't think of what they would want to do post classification walk-through besides maybe export data?

Author: @andf-viam Thank you so much, this is very helpful! Will adapt from this.

linkTitle: "Train and deploy classification models"
weight: 50
type: "docs"
tags: ["data management", "data", "services"]
no_list: true
description: "Use Viam's machine learning capabilities to train image classification models and deploy these models to your machines."
image: "/ml/training.png"
imageAlt: "Machine Learning"
images: ["/ml/training.png"]
---

> Deploy an image classification model onto your machine with Viam's machine learning (ML) service.

Reviewer: Maybe? I think if it just covers deploying that's not covering the full thing but if it covers more it's not just the service
```suggestion
Create and deploy an image classification model onto your machine with Viam's machine learning (ML) capabilities.
```

> Create and deploy an image classification model onto your machine with Viam's machine learning (ML) capabilities.

Reviewer 2: ```suggestion
You can create and deploy an image classification model onto your machine with Viam's machine learning (ML) capabilities.
```

> Do this all on one platform with Viam's data management service: collect data, create a dataset and label it, train or upload an ML model, and deploy the model.

Reviewer: What all? I think this needs wordsmithing

> Then, test if your model works for classifying objects in a camera stream or existing images with the `mlmodel` classification model of vision service.

Author: @npentrel Still might sound a little awkward but I tried to make more classification-centric

> Then, test if your model works for classifying objects in a camera stream or existing images with the `mlmodel` classification model of vision service.

Reviewer: Yep still a little awkward. You put create and deploy in the  first and second sentence. Try and remove that duplication and see if that fixes it.

Author: Done

<table>
  <tr>
    <th>{{<imgproc src="/ml/collect.svg" class="fill alignright" style="max-width: 300px" declaredimensions=true alt="Collect data">}}
      <b>1. Collect</b>

>       <p>Start by collecting data from your cameras, sensors, or any other source on your machine with the <a href="/data/">data management service</a>. You can <a href="/data/view/">view the data</a> on the <b>Data tab</b>.</p>

Reviewer: So this is supposed to be specifically for image classification. I don't think sensors or other sources are needed
```suggestion
      <p>Start by collecting data from your cameras with the <a href="/data/">data management service</a>. You can <a href="/data/view/">view the data</a> on the <b>Data tab</b>.</p>
```

Author: Makes sense, thank you!

    </th>
  </tr>
  <tr>
    <th>{{<imgproc src="/ml/label.svg" class="fill alignleft" style="max-width: 300px" declaredimensions=true alt="Label data">}}

>       <b>2. Create a Dataset and Label</b>

Reviewer 2: ```suggestion
      <b>2. Create a dataset and label</b>
```
[nit] consistency

Author: Done!

>       <p>Once you have collected data, <a href="/data/dataset/">label your data and create a dataset</a> in preparation for training machine learning models.</p>

Reviewer: ```suggestion
      <p>Once you have enough images of the objects you'd like to classify, <a href="/data/dataset/">label your data and create a dataset</a> in preparation for training a classification models.</p>
```

    </th>
  </tr>
  <tr>
    <th>{{<imgproc src="/ml/train.svg" class="fill alignright" style="max-width: 300px" declaredimensions=true alt="Train models">}}
      <b>3. Train an ML model</b>
      <p>Use your labeled data to <a href="/ml/train-model/">train your own models</a> for object classification using data from the <a href="/data/">data management service</a>.</p>
    </th>
  </tr>
  <tr>
    <td>
      <b>4. Deploy your ML model</b>
      <p>To make use of ML models with your machine, use the built-in <a href="/ml/">ML model service</a> to deploy and run the model.</p>
    </td>
  </tr>
  <tr>
    <td>{{<imgproc src="/ml/configure.svg" class="fill alignleft" style="max-width: 300px" declaredimensions=true alt="Configure a service">}}

>       <b>5. Configure a service</b>

Reviewer: ```suggestion
      <b>5. Configure an mlmodel vision service</b>
```

Author: Done

      <p>For object classification, you can use the <a href="/ml/vision/">vision service</a>, which provides an <a href="/ml/vision/classification/#configure-an-mlmodel-classifier">ml model classifier</a> model.</p>
</td>
  </tr>
  <tr>
    <td>{{<imgproc src="ml/deploy.svg" class="fill alignright" style="max-width: 300px" declaredimensions=true alt="Deploy your model">}}
      <b>6. Test your classifier</b>
      <p>Test your <a href="/ml/vision/classification/#test-your-classifier">mlmodel classifier</a> with <a href="/ml/vision/classification/#existing-images-in-the-cloud">existing images in the Viam app</a>, <a href="/ml/vision/classification/#live-camera-footage">live camera footage,</a> or <a href="/ml/vision/classification/#existing-images-on-your-machine">existing images on a computer</a>.</p>
    </td>
  </tr>
</table>

## Next steps

After testing your classifier, see the following to further explore Viam's data management and computer vision capabilities:

- [Export Data Using the Viam CLI](/data/export/): Export your synced data from the Viam cloud.

> - [2D Object Detection](/ml/vision/detection/): configure your machine's camera to draw a bounding box around detected objects, based on a machine learning model.

Reviewer 3: ```suggestion
- [2D Object Detection](/ml/vision/detection/): Configure your machine's camera to draw a bounding box around detected objects, based on a machine learning model.
```

Whoops -- my typo in suggestion!
